drop-in/src/main/java/com/adyen/checkout/dropin/ui/action/ActionComponentDialogFragment.kt

```diff
@@ -147,13 +147,21 @@ class ActionComponentDialogFragment : DropInBottomSheetDialogFragment(), Observe
     private fun getComponent(action: Action): ViewableComponent<*, *, ActionComponentData> {
         val provider =
             getActionProviderFor(action) ?: throw ComponentException("Unexpected Action component type - $actionType")
+
         if (!provider.requiresView(action)) {
             throw ComponentException(
                 "Action is not viewable - action: ${action.type} - " +
                     "paymentMethod: ${action.paymentMethodType}"
             )
         }
-        val component = getActionComponentFor(requireActivity(), provider, dropInViewModel.dropInConfiguration)
+
+        val component = getActionComponentFor(
```

Comment:
I guess this has nothing to do with StatusRepository refactor, so it's better to be done in a separate PR.